exonum-java-binding-core/src/main/java/com/exonum/binding/service/adapters/UserServiceAdapter.java

```diff
@@ -83,16 +84,25 @@ public String getName() {
    * @throws IllegalArgumentException if message is not a valid transaction message of this service
    */
   public UserTransactionAdapter convertTransaction(byte[] transactionMessage) {
-    BinaryMessage message = BinaryMessage.fromBytes(transactionMessage);
-    assert message.getServiceId() == getId() :
-        "Message id is distinct from the service id";
-
-    Transaction transaction = service.convertToTransaction(message);
-    checkNotNull(transaction, "Invalid service implementation: "
-            + "Service#convertToTransaction must never return null.\n"
-            + "Throw an exception if your service does not recognize this message id (%s)",
-        message.getMessageType());
-    return new UserTransactionAdapter(transaction, viewFactory);
+    try {
+      BinaryMessage message = BinaryMessage.fromBytes(transactionMessage);
+      assert message.getServiceId() == getId() :
+          "Message id is distinct from the service id";
+
+      Transaction transaction = service.convertToTransaction(message);
+      checkNotNull(transaction, "Invalid service implementation: "
+              + "Service#convertToTransaction must never return null.\n"
+              + "Throw an exception if your service does not recognize this message id (%s)",
+          message.getMessageType());
+      return new UserTransactionAdapter(transaction, viewFactory);
+    } catch (NullPointerException | IllegalArgumentException e) {
```

Comment:
The NPE above is not a warning, but an error. On the other hand, if a service uses protobuf which allows **any** field to be optional, NPEs are likely in the user code (= and therefore may be regarded as "acceptable" in this case — if your transaction message does not contain a required parameter, than it is not valid). WDYT?

Reply:
I think NPE is "acceptable" in this case. And because of that, I placed `warn`